coap/api/transport.go

```diff
@@ -0,0 +1,236 @@
+package api
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-zoo/bone"
+	"github.com/mainflux/mainflux/coap"
+	"github.com/mainflux/mainflux/coap/nats"
+
+	"math/rand"
+
+	mux "github.com/dereulenspiegel/coap-mux"
+	gocoap "github.com/dustin/go-coap"
+	"github.com/mainflux/mainflux"
+)
+
+var (
+	errBadRequest = errors.New("bad request")
+	errBadOption  = errors.New("bad option")
+	auth          mainflux.ThingsServiceClient
+)
+
+const (
+	maxPktLen = 1500
+	network   = "udp"
+	protocol  = "coap"
+)
+
+const (
+	// Approximately number of supported requests per second
+	timestamp = int64(time.Millisecond) * 31
+)
+
+type handler func(conn *net.UDPConn, addr *net.UDPAddr, msg *gocoap.Message) *gocoap.Message
+
+// NotFoundHandler handles erroneously formed requests.
+func NotFoundHandler(l *net.UDPConn, a *net.UDPAddr, m *gocoap.Message) *gocoap.Message {
+	if m.IsConfirmable() {
+		return &gocoap.Message{
+			Type: gocoap.Acknowledgement,
+			Code: gocoap.NotFound,
+		}
+	}
+	return nil
+}
+
+func version(port string) {
+	b := bone.New()
+	b.GetFunc("/version", mainflux.Version("CoAP"))
+	http.ListenAndServe(port, b)
```

Comment:
You should make two handlers: `HTTP` and `CoAP`, and handle this `ListenAndServe` part in `main`.